packages/api/src/controllers/taskController.js

```diff
@@ -39,6 +39,51 @@ const adminRoles = [1, 2, 5];
 //   return false;
 // };
 
+async function getTaskAssigneeAndFinalWage(farm_id, user_id, task_id) {
+  const {
+    assignee_user_id,
+    assignee_role_id,
+    wage_at_moment,
+    override_hourly_wage,
+  } = await TaskModel.getTaskAssignee(task_id);
+  const { role_id } = await UserFarmModel.getUserRoleId(user_id);
+  if (!canCompleteTask(assignee_user_id, assignee_role_id, user_id, role_id)) {
+    throw new Error("Not authorized to complete other people's task");
+  }
```

Comment:
If we moved this functions to the model, this bit could probably stay in the controller's endpoints even though it's repeated, since it handles sending out the unauthorized response.